esp_encrypted_img/include/esp_encrypted_img.h

```diff
@@ -45,8 +45,14 @@ extern "C" {
 typedef void *esp_decrypt_handle_t;
 
 typedef struct {
-    const char *rsa_pub_key;    /*!< 3072 bit RSA key in PEM format */
-    size_t rsa_pub_key_len;     /*!< Length of the buffer pointed to by rsa_pub_key*/
+    union {
+        const char *rsa_priv_key;                               /*!< 3072 bit RSA private key in PEM format */
+        const char *rsa_pub_key __attribute__((deprecated));    /*!< RSA private key buffer legacy name */
```

Comment:
Lets not add attribute here, it will break the existing releases..

Suggest updating comment that _this name is kept for backward compatibility purpose, but it is not accurate (meaning wise) and hence it would be removed in the next major release_